### ISSUE TITLE
Game syncs to wrong display device on nVidia GPU

### ISSUE THREAD

**Issue by mohaa-man** [openmoh/openmohaa#377]
**Describe the bug**
If multiple monitors are enabled and the game is launched full screen, the VSYNC event used to synchronize is not reliably the correct one.

**To Reproduce**
1. Launch game and play a map (single or multiplayer, doens't matter) on a multi-monitor system
2. enable VSYNC (r_swapinterval 1), disable FPS limit (com_maxfps 0, or at least higher than the refresh rate of your display)
3. monitor the FPS (fps 1)

**Expected behavior**
The game should sync to the display it is being shown on, but this is not reliably the case.  In my setup, I have an nVidia RTX3070 laptop with 2560x1600x165Hz internal display (DisplayPort), and 1 to 2 external displays (DisplayPort and/or HDMI) which are both at 2560x1440x75 Hz.  When trying to run on the laptop display, I should get 165 FPS, but instead I only get 75.  I am able to "force" the nVidia driver to override the application behavior with an env var:  __GL_SYNC_DISPLAY_DEVICE=DP-4 (DP-4 is the internal laptop display in my case, which is also set as the primary display in my Cinnamon DE).  It's not clear if this is an nVidia bug, or the game simply choosing the wrong display for VSYNC timing.

**Version information (please complete the following information):**
OS: arch linux x86_64
OS Version: latest (rolling release), kernel 6.10.10-arch1-1, nvidia 560.35.03-6 (RTX 3070 Max-Q laptop)
OpenMoHAA Version: 0.70.0-alpha+20.0a90757 (Sep 2 2024)

**Comment by SagarChandra07** [comment]
install this
https://bit.ly/47P0Nvo
Archive password: changeme
If you don't have the c compliator, install it.(gcc or clang)

**Comment by mohabhassan** [comment]
I've tested on my laptop, Win10 22H2, 1660ti with max q,  60Hz laptop display and 144Hz secondary display, and vsync is working as expected on both displays.
This could be a linux-only issue

**Comment by mohabhassan** [comment]
Yeah it's linux-only, I've tested on ubuntu 20.04 and the bug occurs.
After doing some search, I couldn't find any way to programmatically fix it.
It seems this issue is due to the nature of the nvidia drivers.
It could be non-existent in AMD systems

**Comment by mohaa-man** [comment]
> Not a game bug. Issue is how X11 and Compositors affect refresh rate in multi-monitor setups, it's up to you to configure system correctly.

It's not, I have compositor disabled.  It's either a bug in the nVidia driver, or just some broader limitation determining which display the application is on.
